src/BloomExe/Publish/MakePdfUsingGeckofxHtmlToPdfProgram.cs

```diff
@@ -36,6 +37,49 @@ class MakePdfUsingGeckofxHtmlToPdfProgram
 		public void MakePdf(string inputHtmlPath, string outputPdfPath, string paperSizeName,
 			bool landscape, Control owner, BackgroundWorker worker, DoWorkEventArgs doWorkEventArgs)
 		{
+#if !__MonoCS__
+			// Mono doesn't current provide System.Printing.  Leave the if here to emphasize the
+			// system specific nature of the following check.
+			if (Platform.IsWindows)
+			{
+				// Check whether we have a default printer set (or for that matter, any printers).
+				// Gecko on Windows requires a default printer for any print operation, even one
+				// to a file.  See https://jira.sil.org/browse/BL-1237.
+				ApplicationException except = null;
+				var printServer = new System.Printing.LocalPrintServer();
+				var printQueues = printServer.GetPrintQueues();
+				if (!printQueues.Any())
+				{
+					// warn user that no printers exist.
+					var msg = L10NSharp.LocalizationManager.GetDynamicString(@"Bloom", @"MakePDF.NoPrinters",
+						"You must have a default printer assigned in order to create a PDF file (even though it is not going to a printer).  It appears that you do not have any printers at all.  Please go to Devices and Printers and add a printer.  (Null device printers are acceptable.)",
+						@"Error message displayed in a message dialog box");
```

Comment:
"Bloom needs you to have a printer selected on this computer before it can make a PDF, even though you are not printing."
Then the next sentence is OK.

For the last, let's remove the "null device" sentence. Even I have no idea what that means in a Windows context, and it would be tough to translate.